### DISCUSSION TITLE
[question] generating ky code from openapi specs

### DISCUSSION THREAD

**Discussion by vonovak** [sindresorhus/ky#362] [Q&A] [answered]
Hi! I just wanted to ask if anyone here is planning to create a tool that would take in openapi definition (yaml from swagger), and spit out typescript code for that api definition, which would be backed by api calls made by `ky`.

Same as https://github.com/OpenAPITools/openapi-generator but for `ky` which is not supported ATM.

Thank you! :)

**Comment by aleclarson** [comment, marked as answer]
You can use https://github.com/acacode/swagger-typescript-api/pull/690 through the `@aleclarson/swagger-typescript-api` package.

**Comment by marcomuser** [comment]
What could be interesting, would be to look into how a ky wrapper could provide type inference from an openapi-typescript generated schema similar to how openapi-fetch is doing it. See: https://openapi-ts.pages.dev/openapi-fetch/

**Reply by marcomuser** [reply]
This now exists (for an equivalent openapi lib): https://heyapi.dev/docs/openapi/typescript/clients/ky 🎉
